static/scenes/railroad/js/present.js

```diff
@@ -50,39 +67,67 @@ class Present {
     }
 
     shoot(targetPosition) {
+        // Move to world space to handle the throw
         this.scene.scene.attach(this.model);
+
+        // new throwing algorithm
+        // specify fixed throwing velocity V_x
+        // calculate maxT to travel linear distance to target
+        // a V_y is chosen given a gravity constant to always look good
+
+        // cache the start and end positions for math later
         this.targetPosition = targetPosition.clone();
-        this.startPosition = new THREE.Vector3();
         this.model.getWorldPosition(this.startPosition);
 
-        var midpoint = this.startPosition.clone()
-            .multiplyScalar(1-midpointT)
-            .addScaledVector(this.targetPosition, midpointT)
-            .add(new THREE.Vector3(0, this.startPosition.distanceTo(this.targetPosition) * midpointOffset));
+        // calculate variables we need to calculate physics stuff
+        var throwDistance = this.startPosition.distanceTo(this.targetPosition);
+        this.durationOfThrow = throwDistance/linearThrowSpeed;
 
-        this.curve = new THREE.QuadraticBezierCurve3(this.startPosition.clone(), midpoint.clone(), this.targetPosition.clone());
+        // if we know gravity, the height delta, and how long the throw takes,
+        // we can choose a start y velocity that will last through the throw
+        this.velocityY = (this.targetPosition.y - this.startPosition.y) / this.durationOfThrow
+            - gravity * this.durationOfThrow / 2;
 
-        this.inFlight = true;
-    }
+        // the predicted linear Vy assuming no gravity
+        const initialVy = this.targetPosition.clone().sub(this.startPosition).normalize().y * linearThrowSpeed;
```

Comment:
I added some logging to double check, and yeah this is the same as just using the `(this.targetPosition.y - this.startPosition.y) / this.durationOfThrow`, already calculated above.

```
:closure.js?module:8 Jez's eq = -6.345463343697578
:closure.js?module:9 initialVy = -6.3454633436975785
:closure.js?module:8 Jez's eq = 0.5083074855737352
:closure.js?module:9 initialVy = 0.5083074855737351
:closure.js?module:8 Jez's eq = -4.909785728905362
:closure.js?module:9 initialVy = -4.909785728905362
:closure.js?module:8 Jez's eq = 0.06645386784523435
:closure.js?module:9 initialVy = 0.06645386784523435
:closure.js?module:8 Jez's eq = -3.77779873567537
:closure.js?module:9 initialVy = -3.77779873567537
:closure.js?module:8 Jez's eq = -0.4738947319894831
:closure.js?module:9 initialVy = -0.4738947319894831
:closure.js?module:8 Jez's eq = 0.6739115767726478
:closure.js?module:9 initialVy = 0.6739115767726478
:closure.js?module:8 Jez's eq = 0.6797564477703062
:closure.js?module:9 initialVy = 0.6797564477703062
```

We can fix that up later though.